src/title.js

```diff
@@ -0,0 +1,441 @@
+		const t = editor.t;
+		const view = editor.editing.view;
+		const viewRoot = view.document.getRoot();
+
+		const bodyPlaceholder = editor.config.get( 'placeholder' ) || t( 'Body' );
+		const titlePlaceholder = t( 'Title' );
+
+		// Attach placeholder to the view title element.
+		editor.editing.downcastDispatcher.on( 'insert:title-content', ( evt, data, conversionApi ) => {
+			enablePlaceholder( {
+				view,
+				element: conversionApi.mapper.toViewElement( data.item ),
+				text: titlePlaceholder
+			} );
+		} );
+
+		// Attach placeholder to first element after a title element and remove it if it's not needed anymore.
+		// First element after title can change so we need to observe all changes keep placeholder in sync.
+		let oldBody;
+
+		// This post-fixer runs after the model post-fixer so we can assume that
+		// the second child in view root will always exist.
+		view.document.registerPostFixer( writer => {
+			const body = viewRoot.getChild( 1 );
+			let hasChanged = false;
+
+			// If body element has changed we need to disable placeholder on the previous element
+			// and enable on the new one.
+			if ( body !== oldBody ) {
+				if ( oldBody ) {
+					hidePlaceholder( writer, oldBody );
+					writer.removeAttribute( 'data-placeholder', oldBody );
+				}
+
+				writer.setAttribute( 'data-placeholder', bodyPlaceholder, body );
+				oldBody = body;
+				hasChanged = true;
+			}
+
+			// Then we need to display placeholder if it is needed.
+			if ( needsPlaceholder( body ) && viewRoot.childCount === 2 && body.name === 'p' ) {
+				hasChanged = showPlaceholder( writer, body ) ? true : hasChanged;
+			// Or hide if it is not needed.
+			} else {
+				hasChanged = hidePlaceholder( writer, body ) ? true : hasChanged;
+			}
+
+			return hasChanged;
+		} );
+	}
+
+	/**
+	 * Creates navigation between Title and Body sections using `Tab` and `Shift+Tab` keys.
+	 *
+	 * @private
+	 */
+	_attachTabPressHandling() {
+		const editor = this.editor;
+		const model = editor.model;
+
+		// Pressing `Tab` inside the title should move the caret to the body.
+		editor.keystrokes.set( 'TAB', ( data, cancel ) => {
+			model.change( writer => {
+				const selection = model.document.selection;
+				const selectedElements = Array.from( selection.getSelectedBlocks() );
+
+				if ( selectedElements.length === 1 && selectedElements[ 0 ].name === 'title-content' ) {
+					const firstBodyElement = model.document.getRoot().getChild( 1 );
+					writer.setSelection( firstBodyElement, 0 );
+					cancel();
+				}
+			} );
+		} );
+
+		// Pressing `Shift+Tab` at the beginning of the body should move the caret to the title.
+		editor.keystrokes.set( 'SHIFT + TAB', ( data, cancel ) => {
+			model.change( writer => {
+				const selection = model.document.selection;
+
+				if ( !selection.isCollapsed ) {
+					return;
+				}
+
+				const root = editor.model.document.getRoot();
+				const selectedElement = Array.from( selection.getSelectedBlocks() )[ 0 ];
```

Comment:
AFAIR there's `first()` util. You can use it here for other similar constructs.

edit: https://github.com/ckeditor/ckeditor5-heading/blob/94665a70bcb3457d94e4c74883bd2fc5d16fe348/src/headingcommand.js#L52